landoapi/landing_worker.py

```diff
@@ -227,88 +246,177 @@ def run_job(self, job, repo, hgrepo, treestatus, patch_bucket):
+                        f"Problem while applying patch in revision {revision_id}:\n\n"
+                        f"{str(exc)}"
+                    )
+                    job.error_breakdown = breakdown
 
-                commit_id = hgrepo.run_hg(["log", "-r", ".", "-T", "{node}"]).decode(
-                    "utf-8"
-                )
-                hgrepo.push(repo.push_path, bookmark=repo.push_bookmark or None)
-        except LostPushRace:
-            logger.info(f"LandingJob {job.id} lost push race, deferring")
-            job.transition_status(
-                LandingJobAction.DEFER,
-                message=f"Lost push race when pushing to {repo.push_path}.",
-                commit=True,
-                db=db,
-            )
-            return False
-        except NoDiffStartLine:
-            logger.exception("Patch without a diff start line.")
-            message = (
-                "Lando encountered a malformed patch, please try again. "
-                "If this error persists please file a bug."
-            )
-            job.transition_status(
-                LandingJobAction.FAIL, message=message, commit=True, db=db
-            )
-            self.notify_user_of_landing_failure(job)
-            return True
-        except PatchConflict as exc:
-            message = (
-                "We're sorry, Lando could not rebase your "
-                "commits for you automatically. Please manually "
-                "rebase your commits and try again.\n\n"
-                f"{str(exc)}"
-            )
-            job.transition_status(
-                LandingJobAction.FAIL, message=message, commit=True, db=db
-            )
-            self.notify_user_of_landing_failure(job)
-            return True
-        except TreeClosed:
-            job.transition_status(
-                LandingJobAction.DEFER,
-                message=f"Tree {repo.tree} is closed - retrying later.",
-                commit=True,
-                db=db,
-            )
-            return False
-        except TreeApprovalRequired:
-            job.transition_status(
-                LandingJobAction.DEFER,
-                message=f"Tree {repo.tree} requires approval - retrying later.",
-                commit=True,
-                db=db,
-            )
-            return False
-        except Exception as e:
-            logger.exception("Unexpected landing error.")
-            job.transition_status(
-                LandingJobAction.FAIL,
-                message=f"An unexpected error occured while landing:\n{e}",
-                commit=True,
-                db=db,
+                    job.transition_status(
+                        LandingJobAction.FAIL, message=message, commit=True, db=db
+                    )
+                    self.notify_user_of_landing_failure(job)
+                    return True
+                except NoDiffStartLine:
+                    # error with patch format? Where does this occur?
```

Comment:
Is this a dev comment?

Reply:
Yes I meant to remove those, will be removing them in next commit.